### PR TITLE
moerr: implement encoding.BinaryUnmarshaler and encoding.BinaryMarsha…

### PR DESCRIPTION
**Pull request by reusee**
…ler for Error

## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [x] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #

## What this PR does / why we need it:

### CODE REVIEW

File: pkg/common/moerr/error.go [hunk 1]
```diff
@@ -349,6 +352,43 @@ func (e *Error) SqlState() string {
...
+func (e *Error) MarshalBinary() ([]byte, error) {
+	ee := encodingErr{
+		Code:      e.code,
+		MysqlCode: e.mysqlCode,
+		Message:   e.message,
+		SqlState:  e.sqlState,
+	}
+	buf := new(bytes.Buffer)
+	if err := gob.NewEncoder(buf).Encode(ee); err != nil {
+		return nil, err
```
Comment (fengttt) [inline pkg/common/moerr/error.go:373]: ConvertGoError(err)

File: pkg/common/moerr/error.go [hunk 2]
```diff
@@ -349,6 +352,43 @@ func (e *Error) SqlState() string {
...
+	}
+	return buf.Bytes(), nil
+}
+
+var _ encoding.BinaryUnmarshaler = new(Error)
+
+func (e *Error) UnmarshalBinary(data []byte) error {
+	var ee encodingErr
+	if err := gob.NewDecoder(bytes.NewReader(data)).Decode(&ee); err != nil {
+		return err
```
Comment (fengttt) [inline pkg/common/moerr/error.go:383]: Here too.

File: pkg/common/moerr/error.go (hunk 1 above)
Comment (reusee) [reply pkg/common/moerr/error.go]: done

File: pkg/common/moerr/error.go (hunk 2 above)
Comment (reusee) [reply pkg/common/moerr/error.go]: done

**Review by fengttt — APPROVED** [review]

**Pull request merged**